quill-cassandra/src/main/scala/io/getquill/source/cassandra/CassandraSourceMacro.scala

```diff
@@ -0,0 +1,51 @@
+package io.getquill.source.cassandra
+
+import scala.reflect.macros.whitebox.Context
+import scala.util.Failure
+import scala.util.Success
+import io.getquill.ast._
+import io.getquill.naming.NamingStrategy
+import io.getquill.source.SourceMacro
+import io.getquill.util.Messages.RichContext
+import io.getquill.util.Show.Shower
+import io.getquill.quotation.Quoted
+import scala.reflect.ClassTag
+import io.getquill.norm.Normalize
+import io.getquill.quotation.IsDynamic
+import io.getquill.source.BindVariables
+import io.getquill.naming.LoadNaming
+
+class CassandraSourceMacro(val c: Context) extends SourceMacro {
+  import c.universe.{ Try => _, Literal => _, Ident => _, _ }
+
+  override protected def prepare(ast: Ast, params: List[Ident]) = {
+    if (!IsDynamic(ast)) {
+      implicit val n = LoadNaming.static(c)(namingType)
+      val (cql, idents) = Prepare(ast, params)
+      c.info(cql)
+      probe(cql)
+      q"($cql, $idents)"
+    } else {
+      c.info("Dynamic query")
+      q"""
+      {
+        implicit val n = ${LoadNaming.dynamic(c)(namingType)}
+        io.getquill.source.cassandra.Prepare($ast, $params)
+      }
+      """
+    }
+  }
+
+  private def probe(cql: String) =
+    resolveSource[CassandraSource[NamingStrategy]].map {
+      _.probe(cql)
+    } match {
+      case Some(Failure(e)) => c.error(s"The sql query probing failed. Reason '$e'")
+      case other            =>
+    }
+
+  private def namingType =
+    c.prefix.actualType
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Usage of head on collections.](https://www.codacy.com/app/fwbrasil/quill/file/1745126947/issues/source?bid=2746728&fileBranchId=3046581#l48)